proposals/0118-partitioned-epoch-reward-distribution.md

```diff
@@ -0,0 +1,275 @@
+---
+simd: '0118'
+title: Partitioned Epoch Rewards Distribution
+authors: Haoran Yi (Anza), Justin Starry (Anza), Tyera Eulberg (Anza)
+category: Standard
+type: Core
+status: Draft
+created: 2024-02-16
+feature: (fill in with feature tracking issues once accepted)
+supersedes: '0015'
+---
+
+## Summary
+
+A new way to distribute epoch rewards across multiple blocks is proposed to
+address the current performance problems associated with epoch reward
+distribution in a first block of a new epoch.
+
+## Motivation
+
+The distribution of epoch rewards at the start block of an epoch becomes a
+significant bottleneck due to the rising number of stake accounts and voting
+nodes on the network.
+
+To address this bottleneck, we propose a new approach for distributing the
+epoch rewards over multiple blocks.
```

Comment:
Just curious - have we measured that it is the distribution (write-back) versus the calculation that is the bottleneck?

Reply:
Yes, definitely, although it doesn't seem like we have that data summarized concisely in any one place. It seems to be mostly in various places in the #proj-epoch-boundary-optimization channel on discord. For instance, here's a comment about the calculation time: https://discord.com/channels/428295358100013066/960593861732884520/1096146390037561414 (64ms for 550K stake accounts)
Whereas distribution is more like 5-10s.